src/utils/generator.js

```diff
@@ -292,6 +299,9 @@ class PossiblyNormalReturnEntry extends GeneratorEntry {
     let alternateEffects =
       completion.alternate instanceof AbruptCompletion ? completion.alternateEffects : empty_effects;
     this.alternateGenerator = Generator.fromEffects(alternateEffects, realm, "AlternateEffects");
+
+    invariant(this.consequentGenerator.isTerminal() === this.alternateGenerator.isTerminal());
```

Comment:
This invariant is not true and I don't see why it should be true. It is perfectly OK and expected for one branch to be a normal completion and the other to be a throw completion.

Reply:
I see there is a PR that will make PossiblyNormalReturnEntry go away. That should erase this concern.